go-controller/pkg/crd/egressfirewall/v1/types.go

```diff
@@ -65,14 +65,16 @@ type EgressFirewallPort struct {
 	Port int32 `json:"port"`
 }
 
-// EgressFirewallDestination is the endpoint that traffic is either allowed or denied to
+// +kubebuilder:validation:MinProperties:=1
```

Comment:
this does look right to me.. I am not sure why this is getting applied to "Type" as well? (I don't see the yaml doing that but that is what seems to be happening) not sure if we should move it to:

// +kubebuilder:validation:MinProperties:=1
// +kubebuilder:validation:MaxProperties:=1
// to is the target that traffic is allowed/denied to
To EgressFirewallDestination `json:"to"`